File: trunk/src/dotnet/yaxgl/YaxglLibrary/Controls/ListBox.cs
using System;
using System.Collections.Generic;
using System.Text;
using System.Xml;

namespace de.yaxgl
{
    public class ListBox : Control
    {
        public ListBox(Container owner, string ID)
        {
            this.owner = owner;
            this.ID = ID;

            this.control = new System.Windows.Forms.ListBox();

            /*register events*/
            this.control.Click += new System.EventHandler(clickEvent);
            ((System.Windows.Forms.ListBox)this.control).SelectedIndexChanged += new System.EventHandler(selectionChangedEvent);
        
        }

        public override void initializeNativeControl(System.Xml.XmlElement xmlElement)
        {
            setBounds(Convert.ToInt32(xmlElement.Attributes["xpos"].InnerText),
                      Convert.ToInt32(xmlElement.Attributes["ypos"].InnerText),
                      Convert.ToInt32(xmlElement.Attributes["width"].InnerText),
                      Convert.ToInt32(xmlElement.Attributes["height"].InnerText));

            if (xmlElement.Attributes["multiselect"].InnerText.Equals("true"))
                setMultiselection(true);
            else if (xmlElement.Attributes["multiselect"].InnerText.Equals("false"))
                setMultiselection(false);


            foreach (XmlNode xmlNode in xmlElement.ChildNodes)
            {
                if (xmlNode.NodeType == XmlNodeType.Element)
                {
                    XmlElement itemElement = (XmlElement)xmlNode;
                    string item = itemElement.Attributes["label"].InnerText;
                    addItem(item);
                    if (itemElement.Attributes["selected"].InnerText.Equals("true"))
                       select(item);
                    
                }
            }
        }


        public void setMultiselection(bool multiselect)
        {
            if (multiselect)
            {
                ((System.Windows.Forms.ListBox)this.control).SelectionMode = System.Windows.Forms.SelectionMode.MultiExtended;

            }
            else
            {
                ((System.Windows.Forms.ListBox)this.control).SelectionMode = System.Windows.Forms.SelectionMode.One;
            }
        }

        public bool isMultiselection()
        {
            if (((System.Windows.Forms.ListBox)this.control).SelectionMode == System.Windows.Forms.SelectionMode.One)
                return false;
            else
                return true;
        }

        public void addItem(string item)
        {
            System.Windows.Forms.ListBox listBox = (System.Windows.Forms.ListBox)this.control;
            listBox.Items.Add(item);
        }

        public void addRange(String[] items)
        {
            System.Windows.Forms.ListBox listBox = (System.Windows.Forms.ListBox)this.control;
            listBox.Items.AddRange(items);
        }

        public void clearItems()
        {
            System.Windows.Forms.ListBox listBox = (System.Windows.Forms.ListBox)this.control;
            listBox.Items.Clear();
        }

        public void select(string selection)
        {
            System.Windows.Forms.ListBox listBox = (System.Windows.Forms.ListBox)this.control;
            if (listBox.Items.Contains(selection))
            {
                listBox.SelectedIndex = listBox.FindStringExact(selection);
            }
        }

        public string[] getSelectedItems()
        {
            System.Windows.Forms.ListBox listBox = (System.Windows.Forms.ListBox)this.control;
            List<string> itemList = new List<string>();
            foreach (string item in listBox.SelectedItems)
            {
                itemList.Add(item);
            }
            return itemList.ToArray();
        }
    
    }
}

File: trunk/src/java/YaxglLibrary/de/yaxgl/Controls/ListBox.java
package de.yaxgl.Controls;

import org.eclipse.swt.SWT;
import org.w3c.dom.Element;
import org.w3c.dom.Node;
import org.w3c.dom.NodeList;

import de.yaxgl.Base.*;
import de.yaxgl.Container.*;

public class ListBox extends Control {

	
	public ListBox(Container owner,String ID)
	{
		this.owner=owner;
		this.ID=ID;
		int style=(SWT.MULTI |SWT.BORDER  | SWT.V_SCROLL);
		
		this.control= new org.eclipse.swt.widgets.List((org.eclipse.swt.widgets.Composite)this.owner.getNativeComponent(),style);
		/*register Events*/
		((org.eclipse.swt.widgets.List)this.control).addSelectionListener(clickEvent(this));
		this.control.addFocusListener(focusEvent(this));
		
		
	}
	
	@Override
	public void initializeNativeControl(Element xmlElement) {
		setBounds(Integer.valueOf(xmlElement.getAttribute("xpos")),
				Integer.valueOf(xmlElement.getAttribute("ypos")),
				Integer.valueOf(xmlElement.getAttribute("width")),
				Integer.valueOf(xmlElement.getAttribute("height")));
		
		NodeList nodeList=xmlElement.getChildNodes();
    
		for(int i=0;i<nodeList.getLength();i++)
		{
           if (nodeList.item(i).getNodeType() == Node.ELEMENT_NODE)
           {
               Element itemElement = (Element)nodeList.item(i);
               addItem(itemElement.getAttribute("label"));
               if(itemElement.hasAttribute("selected"))
            	   select(itemElement.getAttribute("label"));
           }
       }
      
	}
	
	
	public void addItem(String item)
    {
		((org.eclipse.swt.widgets.List)this.control).add(item);
    }

    public void addRange(String[] items)
    {
    	((org.eclipse.swt.widgets.List)this.control).setItems(items);
    }

    public void clearItems()
    {
    	((org.eclipse.swt.widgets.List)this.control).removeAll();
    }

    public void select(String selection)
    {
    	int index=((org.eclipse.swt.widgets.List)this.control).indexOf(selection);
    	((org.eclipse.swt.widgets.List)this.control).select(index);
    }

    public String[] getSelectedItems()
    {
        return ((org.eclipse.swt.widgets.List)this.control).getSelection();
    }


}
